streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -716,7 +716,7 @@ public void process(final Record<Integer, Integer> record) {
                 final SessionStore<Integer, Integer> stateStore =
                     context().getStateStore(sessionStoreStoreBuilder.name());
                 stateStore.put(
-                    new Windowed<>(record.key(), new SessionWindow(WINDOW_START, WINDOW_START)),
+                    new Windowed<>(record.key(), new SessionWindow((record.timestamp() / WINDOW_SIZE.toMillis()) * WINDOW_SIZE.toMillis(), (record.timestamp() / WINDOW_SIZE.toMillis()) * WINDOW_SIZE.toMillis() + WINDOW_SIZE.toMillis())),
```

Comment:
So it should be `new SessionWindow((record.timestamp() / WINDOW_SIZE.toMillis()) * WINDOW_SIZE.toMillis(), (record.timestamp() / WINDOW_SIZE.toMillis()) * WINDOW_SIZE.toMillis() + WINDOW_SIZE.toMillis())`?
but they are same the session window size seem to become 0
```
 /**
     * Create a new window for the given start time and end time (both inclusive).
     *
     * @param startMs the start timestamp of the window
     * @param endMs   the end timestamp of the window
     * @throws IllegalArgumentException if {@code startMs} is negative or if {@code endMs} is smaller than {@code startMs}
     */
    public SessionWindow(final long startMs, final long endMs) throws IllegalArgumentException {
        super(startMs, endMs);
    }
```
so you mean, each session window size is 0 and they interval is window_size, and in the end each record just falls in their session window.

Reply:
I think it should be `new SessionWindow(record.timestamp(), record.timestamp())`

> but their session window size seem to become 0

Yes, the size would become `1` (note, for sessions window, both upper and lower bound are inclusive in contrast to Tumbling/Hopping window). For session windows, there is no fixed size, but the "inactivity gap" parameter (the test only re-uses `WINDOW_SIZE` variable as "gap" -- this might be a little bit confusing). The size of the created windows depend on the data -- in the current test setup, records are apart from each other more than the "gap" and thus each record creates it's own session, and thus each session has size `1` -- only if two (or more) records fall into the same session (ie, their timestamp are close enough to each other, ie, `<= gap` the session size would become larger).